docs/concepts/components/children.md

```diff
@@ -0,0 +1,255 @@
+---
+title: Children
+description: 
+---
+
+## General usage
+
+_Most of the time,_ when allowing a component to have children, you don't care 
+what type of children the component has. In such cases, the below example will
+suffice.
+
+```rust
+use yew::prelude::*;
+
+#[derive(Properties, Clone)]
+pub struct ListProps {
+	#[prop_or_default]
+	pub children: Children,
+}
+
+pub struct List {
+	props: ListProps,
+}
+
+impl Component for List {
+	type Properties = ListProps;
+	// ...
+
+	fn view(&self) -> Html {
+		html! {
+			<div class="list">
+				{ for self.props.children.iter() }
+			</div>
+		}
+	}
+}
+```
+
+## Advanced usage
+
+There are some cases where you wouldn't want to use the `Children` type
+and instead use an alternative method:
+
+ - You want to only allow components of a specific type to be used as children for your component.
+
+ - You want to share state between a component and its children.
+
+### Typed children
+
+In cases where you want one type of component to be passed as children to your component,
+you can use `yew::html::ChildrenWithProps<T>`.
+
+```rust
+use yew::prelude::*;
+use yew::html::ChildrenWithProps;
+
+// ...
+
+#[derive(Properties, Clone)]
+pub struct ListProps {
+	#[prop_or_default]
+	pub children: ChildrenWithProps<Item>,
+}
+
+pub struct List {
+	props: ListProps,
+}
+
+impl Component for ListProps {
+	type Properties = ListProps;
+	// ...
+
+	fn view(&self) -> Html {
+		html! {
+			<div class="list">
+				{ for self.props.children.iter() }
+			</div>
+		}
+	}
+}
+```
+
+Of course, sometimes you might need to restrict the children to a few different
+components. In these cases, you have to get a little more hands-on with Yew.
+
+The `derive_more` crate is used here for better ergonomics. If you don't want
+to use it, replace `ItemPropVariants` with this:
+
+```rust
+#[derive(Clone)]
+pub enum ItemPropVariants {
+	MyFirstComponent(MyFirstComponentProps),
+	MySecondComponent(MySecondComponentProps),
+}
+
+impl From<MyFirstComponentProps> for ItemPropVariants {
+	fn from(props: MyFirstComponentProps) -> Self {
+		Self::MyFirstComponent(props)		
+	}
+}
+
+impl From<MySecondComponentProps> for ItemPropVariants {
+	fn from(props: MySecondComponentProps) -> Self {
+		Self::MySecondComponent(props)
+	}
+}
+```
```

Comment:
I feel like it's enough to leave it at that.
```suggestion
to use it, you can manually implement `From` for each variant.
```
Perhaps we should also link to the documentation for `derive_more::From` to encourage users to learn about what it does there?